src/renderer/SpeechBubbleSkin.ts

```diff
@@ -24,27 +39,32 @@ export default class SpeechBubbleSkin extends Skin {
     this.height = 0;
     this.offsetX = -bubbleStyle.strokeWidth / 2;
     this.offsetY = this.offsetX + bubbleStyle.tailHeight;
-
-    this._renderBubble(this._bubble);
   }
 
   // To ensure proper text measurement and drawing, it's necessary to restyle the canvas after resizing it.
-  _restyleCanvas() {
-    const ctx = this._canvas.getContext("2d");
+  private _restyleCanvas(): void {
+    const ctx = this._ctx;
     ctx.font = "16px sans-serif";
     ctx.textBaseline = "hanging";
   }
 
-  set flipped(flipped) {
+  public set flipped(flipped: boolean) {
     this._flipped = flipped;
     this._rendered = false;
   }
 
-  _renderBubble(bubble, scale) {
+  private _renderBubble(bubble: SpeechBubble, scale: number): void {
     const canvas = this._canvas;
-    const ctx = canvas.getContext("2d");
-
-    const renderBubbleBackground = (x, y, w, h, r, style) => {
+    const ctx = this._ctx;
+
+    const renderBubbleBackground = (
+      x: number,
+      y: number,
+      w: number,
+      h: number,
+      r: number,
+      style: "say" | "think"
```

Comment:
Can we enumalize this, maybe import the bubble style from its definition I think in `Sprite.ts`?

Reply:
Making it an enum would be a breaking change, I think. I've exported the type as a union.

Reply:
Thanks, yeah, I've put it into a separate slightly broader issue with other similar breaking changes: #169